app/controllers/api/user_books_controller.rb

```diff
@@ -49,7 +49,10 @@ def user_book_params
   end
 
   def serialize_data(data)
-    options = { include: [:book] }
-    return UserBookSerializer.new(data, options).serializable_hash[:data].pluck(:attributes)
+    return UserBookSerializer.new(data).serializable_hash[:data].pluck(:attributes)
   end
-end
\ No newline at end of file
+  
+  def serialize(data)
+    return UserBookSerializer.new(data).serializable_hash[:data][:attributes]
+  end
```

Comment:
what's the difference between serialzie and serialize_data? Do you need both?

Reply:
the serialize_data is for the 'index' method which returns an array of objects. the reason why its just plucking the :attributes from the array.

the serialize is for the other methods like 'create', 'update', 'show' which returns a single object. the reason why [:attributes] can be accessed directly here.

serialize_data won't work on single objects and serialize won't work on an array. the reason why I felt to create a seperate one. I tried making a function that would handle both cases but I was unsuccessful.

actually, now that I think about it, I can also do it like this 
![image](https://user-images.githubusercontent.com/91857581/190117610-45832bc0-cff8-4873-893f-0a8f53f5899e.png)
where I just call that specific serializer on that index method. this way, I don't have to create a function for that since its only getting used once.

then I'll just leave the serialize function for the other methods in the controller
![image](https://user-images.githubusercontent.com/91857581/190118113-275864a5-4443-4018-af20-ab146e8a51bf.png)

is this better? or is there another way this can be handled?